conan/cli/commands/remove.py

```diff
@@ -27,26 +50,44 @@ def remove(conan_api: ConanAPI, parser, *args):
                              "os=Windows AND (arch=x86 OR compiler=gcc)")
     parser.add_argument('-r', '--remote', action=OnceArgument,
                         help='Will remove from the specified remote')
+    parser.add_argument("-l", "--list", help="Package list file")
     args = parser.parse_args(*args)
 
+    if args.pattern is None and args.list is None:
+        raise ConanException("Missing pattern or package list file")
+    if args.pattern and args.list:
+        raise ConanException("Cannot define both the pattern and the package list file")
+    if args.package_query and args.list:
+        raise ConanException("Cannot define package-query and the package list file")
+
     ui = UserInput(conan_api.config.get("core:non_interactive"))
     remote = conan_api.remotes.get(args.remote) if args.remote else None
 
     def confirmation(message):
         return args.confirm or ui.request_boolean(message)
 
-    ref_pattern = ListPattern(args.pattern, rrev="*", prev="*")
-    select_bundle = conan_api.list.select(ref_pattern, args.package_query, remote)
-
-    if ref_pattern.package_id is None:
-        if args.package_query is not None:
+    if args.list:
+        listfile = make_abs_path(args.list)
+        multi_package_list = MultiPackagesList.load(listfile)
+        package_list = multi_package_list["Local Cache" if not remote else remote.name]
```

Comment:
Maybe time to move "Local Cache" to some kind of constant? (Not on this PR) It's in the codebase a few times by now